File: C++/Problems/Problems/problem_1848.cpp
#include "problem_1848.h"

bool problem_1848::test() {
    std::vector nums  { 1, 2, 3, 4, 5 };
    int target = 5;
    int start = 3;

    int expected = 1;

    auto result = getMinDistance(nums, target, start);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1848::getMinDistance(std::vector<int> &nums, int target, int start) {
    int n = nums.size();
    int result = INT_MAX;
    for (int i = 0; i < n; ++i)
    {
        if (nums[i] == target)
            result = std::min(result, std::abs(i - start));
    }

    return result;
}

File: Python/Problems/Problems/problem1848.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1848(Problem):
    def test(self) -> bool:
        nums = [1, 2, 3, 4, 5]
        target = 5
        start = 3

        expected = 1

        result = self.getMinDistance(nums, target, start)

        print(result)

        return result == expected

    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        n = len(nums)
        result = sys.maxsize
        for i in range(n):
            if nums[i] == target:
                result = min(result, abs(i - start))

        return result